test/extensions/dynamic_modules/http/integration_test.cc

```diff
@@ -606,25 +606,75 @@ TEST_P(DynamicModulesIntegrationTest, StatsCallbacks) {
   }
 }
 
-TEST_P(DynamicModulesIntegrationTest, InjectBody) {
-  initializeFilter("inject_body");
-  codec_client_ = makeHttpConnection(makeClientConnection((lookupPort("http"))));
+std::string terminal_filter_config;
 
-  auto response =
-      codec_client_->makeRequestWithBody(default_request_headers_, "ignored_request_body");
-  waitForNextUpstreamRequest();
-  upstream_request_->encodeHeaders(default_response_headers_, false);
-  upstream_request_->encodeData("ignored_response_body", true);
+class DynamicModulesTerminalIntegrationTest
+    : public testing::TestWithParam<Network::Address::IpVersion>,
+      public HttpIntegrationTest {
+public:
+  DynamicModulesTerminalIntegrationTest()
+      : HttpIntegrationTest(Http::CodecType::HTTP2, GetParam(), terminal_filter_config) {};
+
+  static void SetUpTestSuite() { // NOLINT(readability-identifier-naming)
```

Comment:
I followed the pattern of EchoIntegrationTest for a terminal filter but let me know if there are any better patterns for setting up a config for a filter chain with no upstream.